src/kinematics_plugin.cpp

```diff
@@ -253,6 +253,12 @@ struct BioIKKinematicsPlugin : kinematics::KinematicsBase {
                 std::string("bio2_memetic"));
     lookupParam("counter", ikparams.enable_counter, false);
     lookupParam("threads", ikparams.thread_count, 0);
+    ros::NodeHandle nodeHandle("~");
+    if (nodeHandle.hasParam("random_seed")) {
+      int random_seed;
+      lookupParam("random_seed", random_seed, 0);
+      ikparams.random_seed = random_seed;
+    }
```

Comment:
Thanks for updating the function call, but you still have the same bug. :)
If the user specifies `random_seed` in `kinematics.yaml`, the parameter ends up in `robot_description_kinematics/<group_name>/random_seed`, but your logic skips it in this case.

I believe you can simply replace the whole block by something like
```
lookupParam("random_seed", random_seed, static_cast<int>(std::random_device()()));
```
if you change the type of `random_seed` to `int` and leave out the initialization in the `IKParams` datastructure.
Alternatively something like
```
int seed;
lookupParam("random_seed", seed, static_cast<int>(std::random_device()()));
random_seed = static_cast<std::minstd_rand::result_type>(seed);

Reply:
done.